lighthouse-core/gather/gatherers/accessibility.js

```diff
@@ -63,20 +59,14 @@ class Accessibility extends Gatherer {
     })()`;
 
     return driver
-        .evaluateAsync(expression)
-        .then(returnedValue => {
-          if (!returnedValue) {
-            return Accessibility._errorAccessibility('Unable to parse axe results');
-          }
-
-          if (returnedValue.error) {
-            return Accessibility._errorAccessibility(returnedValue.error);
-          }
+      .evaluateAsync(expression)
+      .then(returnedValue => {
+        if (!returnedValue || !Array.isArray(returnedValue.violations)) {
```

Comment:
since we're nuking all the remaining checks on the violations array, lets just add an assertion in `accessibility-test.js` that we throw in this case.